src/database/db.js

```diff
@@ -215,34 +218,45 @@ export const getBookmarksForID = (baniId, length, language) => {
       baniLength = EXISTS_MEDIUM;
   }
 
-  return new Promise((resolve) => {
-    initDB().then((db) => {
-      db.transaction((tx) => {
-        tx.executeSql(
-          `SELECT ID, BaniShabadID, Seq, Gurmukhi, Transliterations, TukGurmukhi, TukTransliterations FROM Banis_Bookmarks WHERE Bani = ${baniId} AND BaniShabadID in (SELECT ID from mv_Banis_Shabad where Bani = ${baniId} AND ${baniLength} = 1)` +
-            ` ORDER BY Seq ASC;`,
-          [],
-          (_tx, results) => {
-            const totalResults = new Array(results.rows.length);
-            const len = results.rows.length;
-            for (let i = 0; i < len; i += 1) {
-              const row = results.rows.item(i);
-              const { BaniShabadID, Gurmukhi, Transliterations, TukGurmukhi, TukTransliterations } =
-                row;
-              totalResults[i] = {
-                shabadID: BaniShabadID,
-                gurmukhi: Gurmukhi,
-                tukGurmukhi: TukGurmukhi,
-                translit: getTranslitText(Transliterations, language),
-                tukTranslit: TukTransliterations
-                  ? getTranslitText(TukTransliterations, language)
-                  : null,
-              };
+  return new Promise((resolve, reject) => {
+    initDB()
+      .then((db) => {
+        db.transaction((tx) => {
+          tx.executeSql(
+            `SELECT ID, BaniShabadID, Seq, Gurmukhi, Transliterations, TukGurmukhi, TukTransliterations FROM Banis_Bookmarks WHERE Bani = ${baniId} AND BaniShabadID in (SELECT ID from mv_Banis_Shabad where Bani = ${baniId} AND ${baniLength} = 1)` +
+              ` ORDER BY Seq ASC;`,
+            [],
+            (_tx, results) => {
+              const totalResults = new Array(results.rows.length);
+              const len = results.rows.length;
+              for (let i = 0; i < len; i += 1) {
+                const row = results.rows.item(i);
+                const {
+                  BaniShabadID,
+                  Gurmukhi,
+                  Transliterations,
+                  TukGurmukhi,
+                  TukTransliterations,
+                } = row;
+                totalResults[i] = {
+                  shabadID: BaniShabadID,
+                  gurmukhi: Gurmukhi,
+                  tukGurmukhi: TukGurmukhi,
+                  translit: getTranslitText(Transliterations, language),
+                  tukTranslit: TukTransliterations
+                    ? getTranslitText(TukTransliterations, language)
+                    : null,
+                };
+              }
```

Comment:
Instead of for loop, we can use `Array.from`

```javascript
const totalResults = Array.from({ length: results.rows.length }, (_, i) => {
   const row = results.rows.item(i);

   return {
      shabadID: row.BaniShabadID,
      gurmukhi: row.Gurmukhi,
      tukGurmukhi: row.TukGurmukhi,
      translit: getTranslitText(row.Transliterations, language),
      tukTranslit: row.TukTransliterations ? 
                   getTranslitText(row.TukTransliterations, language)
                    : null,
     };
});
```

Reply:
fixed